pyomo/repn/plugins/nl_writer.py

```diff
@@ -2099,12 +2180,11 @@ def exitNode(self, node, data):
         #
         # General expressions...
         #
-        if all(arg[0] is _CONSTANT for arg in data):
-            return (
-                _CONSTANT, node._apply_operation(list(map(
-                    itemgetter(1), data)))
-            )
-        return self._operator_handles[node.__class__](self, node, *data)
+        if node.is_potentially_variable():
+            cls = node.__class__
+        else:
+            cls = node.potentially_variable_base_class()
```

Comment:
Instead of adding a new method to the `NPV_Mixin` class, could we add the NPV classes as keys to `self._operator_handles` that just map to the same functions as the non-NPV classes?

Reply:
Good call.  Done.

Reply:
Does this change not remove the need for the new `potentially_variable_base_class` method?